File: FCloud3.Entities/Messages/OpRecord.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FCloud3.Entities.Messages
{
    public class OpRecord : IDbModel
    {
        public int Id { get; set; }
        public string? Content { get; set; }
        public OpRecordOpType OpType { get; set; }
        public OpRecordTargetType TargetType { get; set; }
        /// <summary>
        /// 主Id（例如“为目录加入词条”的目录Id），对应当前操作目标（TargetType）
        /// </summary>
        public int ObjA { get; set; }
        /// <summary>
        /// 副Id（例如“为目录加入词条”的词条Id）
        /// </summary>
        public int ObjB { get; set; }
        public int CreatorUserId { get; set; }
        public DateTime Created { get; set; }
        public DateTime Updated { get; set; }
        public bool Deleted { get; set; }
    }
    public enum OpRecordOpType
    {
        None = 0,
        Create = 10,
        Edit = 20,
        EditImportant = 21,
        Remove = 30,
    }
    public enum OpRecordTargetType
    {
        None = 0,
        WikiItem = 10,
        WikiPara = 11,
        TextSection = 20,
        FreeTable = 21,
        FileDir = 30,
        FileItem = 31,
        UserGroup = 40,
        User = 41
    }
}

File: FCloud3.AppFront/FCloud3Front/src/models/messages/opRecord.ts
export interface OpRecordViewModel
{
    Id: number
    Content: string
    OpType: OpRecordOpType
    TargetType: OpRecordTargetType
    TargetObjId: number
    OtherObjId: number
    UserId: number
    UserName: number
    Time: string
}
export enum OpRecordOpType
{
    None = 0,
    Create = 10,
    Edit = 20,
    EditImportant = 21,
    Remove = 30,
}
export enum OpRecordTargetType
{
    None = 0,
    WikiItem = 10,
    WikiPara = 11,
    TextSection = 20,
    FreeTable = 21,
    FileDir = 30,
    FileItem = 31,
    UserGroup = 40,
    User = 41
}

export interface OpRecordGetRequest{
    Skip: number
    User: number
}

export function OpTypeReadable(t:OpRecordOpType){
    if(t == OpRecordOpType.Create)
        return "创建"
    if(t == OpRecordOpType.Edit)
        return "编辑"
    if(t == OpRecordOpType.EditImportant)
        return "变动"
    if(t == OpRecordOpType.Remove)
        return "删除"
}
export function OpTypeColor(t:OpRecordOpType){
    if(t == OpRecordOpType.Create)
        return "green"
    if(t == OpRecordOpType.Edit)
        return "cornflowerblue"
    if(t == OpRecordOpType.EditImportant)
        return "fuchsia"
    if(t == OpRecordOpType.Remove)
        return "red"
}
export function TargetTypeReadable(t:OpRecordTargetType){
    if(t == OpRecordTargetType.WikiItem)
        return "词条"
    if(t == OpRecordTargetType.WikiPara)
        return "段落"
    if(t == OpRecordTargetType.TextSection)
        return "文本段"
    if(t == OpRecordTargetType.FreeTable)
        return "表格"
    if(t == OpRecordTargetType.FileDir)
        return "目录"
    if(t == OpRecordTargetType.FileItem)
        return "文件"
    if(t == OpRecordTargetType.UserGroup)
        return "用户组"
    if(t == OpRecordTargetType.User)
        return "用户"
}